### PR TITLE
Refactor: move initial state to props for CatalogGraphPage

### PR DESCRIPTION
**Pull request by shubhamjajoo**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by shubhamjajoo** [comment]
@sennyeya As per your comment on this [issue](https://github.com/backstage/backstage/issues/17096#issuecomment-1500457684), created this PR to refactor props.
Can you pls start reviewing this ?

File: .changeset/blue-ants-scream.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-graph': minor
+---
+
+Refactor `props` of CatalogGraphPage component. Move properties from `initialState` to `props` and use from `EntityRelationsGraphProps`.
```
Comment (benjdlambert) [inline .changeset/blue-ants-scream.md:5]: I think that you might want to elaborate that it's a breaking change here, and provide some examples for how to migrate to the new version if using the `initialState` version as it is today.

`**BREAKING**: ...` is normally how we prefix these changes too.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Hey :wave: think this is looking good - we just want to nail down the changeset I think. 

Also would be good to get some feedback from @rugvip and @sennyeya too

File: plugins/catalog-graph/src/components/CatalogGraphPage/useCatalogGraphPage.ts [hunk 2]
```diff
@@ -56,19 +56,13 @@ export type CatalogGraphPageValue = {
 
 export function useCatalogGraphPage({
   initialState = {},
+  ...props
 }: {
   initialState?: {
     selectedRelations?: string[] | undefined;
     selectedKinds?: string[] | undefined;
-    rootEntityRefs?: string[];
```
Comment (sennyeya) [inline plugins/catalog-graph/src/components/CatalogGraphPage/useCatalogGraphPage.ts:63]: I may have been a bit overzealous in my original comment. I think this use of `initialState` makes sense for `useCatalogGraphPage` and _probably_ `CatalogGraphPage` as well. I didn't realize we were supporting dynamically setting all of these options.

@benjdlambert is this generally how `initialState` is used across the project?

Comment (sennyeya) [reply plugins/catalog-graph/src/components/CatalogGraphPage/useCatalogGraphPage.ts:63]: @benjdlambert bump.

**Pull request closed without merging**